pkg/scd/operational_intents_handler.go

```diff
@@ -490,6 +511,14 @@ func (a *Server) upsertOperationalIntentReference(ctx context.Context, authorize
 					}
 				}
 
+				removePreviousImplicitSubscription, err = subscriptionIsOnlyAttachedToOIR(ctx, r, &id, previousSubscriptionID)
```

Comment:
Any reason why not deleting the subscription in this block rather than setting a flag? That would simplify the control flow.

Reply:
At this point in the transaction the subscription cannot be deleted, because it is referenced by the operational intent. 

The previous implicit subscription can only be removed once the OIR has been updated to point to another subscription.

Reply:
Note to self: will need to confirm the above again, as I might have been confused by another error.